### PR TITLE
Asset Owner OMAS fixes

### PR DESCRIPTION
**Pull request by cmgrote**
Fixes issues identified in Asset Owner OMAS while trying to make use of the APIs directly, including #3568

### CODE REVIEW

File: open-metadata-implementation/common-services/ocf-metadata-management/ocf-metadata-handlers/src/main/java/org/odpi/openmetadata/commonservices/ocf/metadatamanagement/handlers/SchemaTypeHandler.java [hunk 1]
```diff
@@ -1059,7 +1059,7 @@ else if (instanceElementType.getElementOrigin() == ElementOrigin.LOCAL_COHORT)
             return true;
         }
 
-        return false;
+        return instanceElementType.getElementOrigin() == ElementOrigin.CONFIGURATION;
```
Comment (mandy-chessell) [inline open-metadata-implementation/common-services/ocf-metadata-management/ocf-metadata-handlers/src/main/java/org/odpi/openmetadata/commonservices/ocf/metadatamanagement/handlers/SchemaTypeHandler.java:1062]: Configuration is not local cohort

File: open-metadata-implementation/common-services/ocf-metadata-management/ocf-metadata-handlers/src/main/java/org/odpi/openmetadata/commonservices/ocf/metadatamanagement/handlers/SchemaTypeHandler.java [hunk 2]
```diff
@@ -1395,13 +1395,25 @@ public SchemaAttributeBuilder  getSchemaAttributeBuilder(SchemaAttribute schemaA
      *
      * @param schemaAttribute bean to interrogate
      * @return guid of the type
+     * @throws InvalidParameterException if the provided schema attribute type cannot be resolved
      */
-    private String getSchemaAttributeTypeGUID(SchemaAttribute   schemaAttribute)
+    private String getSchemaAttributeTypeGUID(SchemaAttribute   schemaAttribute) throws InvalidParameterException
```
Comment (mandy-chessell) [inline open-metadata-implementation/common-services/ocf-metadata-management/ocf-metadata-handlers/src/main/java/org/odpi/openmetadata/commonservices/ocf/metadatamanagement/handlers/SchemaTypeHandler.java:1400]: The typeGUID should be set where the SchemaAttribute object is created.  Isn't there a method like createEmptySchemaAtttibute?  This seems too late.

**Review by mandy-chessell — CHANGES_REQUESTED** [review]

File: open-metadata-implementation/common-services/ocf-metadata-management/ocf-metadata-handlers/src/main/java/org/odpi/openmetadata/commonservices/ocf/metadatamanagement/handlers/SchemaTypeHandler.java (hunk 1 above)
Comment (cmgrote) [reply open-metadata-implementation/common-services/ocf-metadata-management/ocf-metadata-handlers/src/main/java/org/odpi/openmetadata/commonservices/ocf/metadatamanagement/handlers/SchemaTypeHandler.java:1062]: Configuration is the default value when no other value is provided -- and unfortunately without this an external sourced entity creation is attempted (without any external metadata collection ID)...  For some reason this logic differentiating between external and local cohort is only present when trying to create schema attributes, and not schema types (not sure why they're different in this regard)?

File: open-metadata-implementation/common-services/ocf-metadata-management/ocf-metadata-handlers/src/main/java/org/odpi/openmetadata/commonservices/ocf/metadatamanagement/handlers/SchemaTypeHandler.java (hunk 2 above)
Comment (cmgrote) [reply open-metadata-implementation/common-services/ocf-metadata-management/ocf-metadata-handlers/src/main/java/org/odpi/openmetadata/commonservices/ocf/metadatamanagement/handlers/SchemaTypeHandler.java:1400]: The `SchemaAttribute` object is created by the `clone()` method of the `SchemaAttributeProperties` bean (see line 238 of `AssetOwnerRESTServices`)...  Not sure how I could do this other than extending the bean itself (to include the typedef GUID), or passing through the repository helper to the bean to be able to lookup the typedef GUID (?)

The actual persistence of the `SchemaAttribute` object isn't until after the current point anyway (line 248 of `AssetOwnerRESTServices`), so any exception should break us out before persistence of either the schema type or its attributes...

**Comment by planetf1** [comment]
@cmgrote is the intent still to merge, I notice it's been idle for 6 weeks now. Some remerging will be needed with the recent changes

**Comment by cmgrote** [comment]
I think the intention was always that some parallel changes @mandy-chessell was making would replace the need for this PR, but have lost track of what the current state is on that / those (?)

**Comment by planetf1** [comment]
@mandy-chessell  ?

**Comment by mandy-chessell** [comment]
I believe it is fixed

**Comment by planetf1** [comment]
@cmgrote Agree? Can we close this now?

**Pull request closed without merging**